NLSY/nlsy_imputation.qmd

```diff
@@ -0,0 +1,227 @@
+---
+title: "NLSY Parent Income and Wealth Imputation"
+format:
+  html:
+    embed-resources: true
+editor: visual
+execute: 
+  warning: false
+fig-cap-location: top
+toc: true
+toc-expand: true
+---
+
+## Data Cleaning
+
+```{r}
+#Packages
+librarian::shelf(dplyr, mice, car, corrplot, caret, tidyr)
+source(here::here("NLSY/nlsy_lib.R"))
+
+# Read data
+nlsydf = readRDS(paste0(here::here(), "/NLSY/NLSY-college-finance.rds"))
+```
+
+```{r}
+test <- nlsydf %>% #Add additional variables that should be considered during multiple imputation
+    select(id = "PUBID_1997",
+           mom_age_birth = "CV_BIO_MOM_AGE_YOUTH_1997", # numeric
+           age_resp =  "CV_AGE_12/31/96_1997",
+           hispanic = "KEY_ETHNICITY_1997", # binary
+           race = "KEY_RACE_1997", # categorical
+           mom_education = "CV_HGC_RES_MOM_1997", # categorical - highest grade completed in 1997 by mom
+           dad_education = "CV_HGC_RES_DAD_1997", # categorical - highest grade completed in 1997 by dad
+           savings = "P5-130_1997", #binary - yes/no retirement savings
+           home = "P5-101_1997", # binary - homeownership or renting
+           both_parents = "YOUTH_BOTHBIO.01_1997",
+           par1_deceased = "YOUTH_NONR1DEAD.01_1997",
+           par2_deceased = "YOUTH_NONR2DEAD.01_1997",
+           pincome = "CV_INCOME_GROSS_YR_1997",
+           pnetworth = "CV_HH_NET_WORTH_P_1997",
+           # Also, it seems that this represents the wealth of the household in which a respondent lives. That means that for a respondent who lives with only one parent, the net-worth variable represents the wealth of that parent.
+           # In the case of an independent respondent, contains respondent's, rather than parents', wealth.
+    ) %>%
+    mutate(mom_age = mom_age_birth + age_resp)
+
+# Making dummy or factor variables
+test <- test |>
+    mutate(
+        mom_educ_hs = factor(nlsy_encode_educ4(mom_education)),
+        dad_educ_hs = factor(nlsy_encode_educ4(dad_education)),
+        race = factor(race),
+        hisp = factor(hispanic),
+        race_eth = ifelse(hisp == "Yes" & race == "White", "White Hispanic",  NA), # Race and ethnicity interaction
+        race_eth = ifelse(hisp == "No" & race == "White", "White NonHispanic", race_eth),
+        race_eth = ifelse(hisp == "Yes" & race == "Black or African American", "Black Hispanic", race_eth),
+        race_eth = ifelse(hisp == "No" & race == "Black or African American", "Black NonHispanic", race_eth),
+        race_eth = ifelse(hisp == "Yes" & race == "American Indian, Eskimo, or Aleut", "NonBW Hispanic", race_eth),
```

Comment:
In my previous comment about coding race and ethnicity, I suggested using all the race categories present in the sample because it would increase the predictive power of this variable. Knowing that a person is American Indian or Asian likely provides more information about the person's income and wealth than if know only that the person is NonBW.

Reply:
Good flag!! I was still thinking about this from a missingness concern, which is not applicable here. Updated!